query-request.hh

```diff
@@ -239,11 +239,16 @@ public:
 };
 
 // See docs/design-notes/reverse-reads.md
+// In the following functions, `schema` may be reversed or not (both work).
+// FIXME: it would be better to have functions which don't require schema at all...
```

Comment:
We require the schema because we use `slice_builder`. Removing the schema would require either over-complicating the slice builder, or reverting to allowing anybody to mutate slice internals. We don't want either.

Reply:
I don't like that because when we have half of the mutating methods on the slice and the other half on the slice builder. Better have all of them in one place.

Reply:
Back when I implemented this commit, 100% of the methods were on the slice :P
I left this FIXME with the thought that maybe I'll convert the approach back to the slice-method-based one eventually.
But maybe I'll just be lazy and remove it.

Reply:
I mean *all* the mutating methods, not just the reversing related ones. And we have the slice builder for a long time now.